### ISSUE TITLE
android: black screen in Android Emulator on Linux

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#1035]
A coworker (@DentonGentry) was trying to run the Tailscale Android app in the Android Emulator on a remote Linux machine (RDP/VNCing into it somehow) and only got a black screen.

Same OpenGL bug that has affected other devices, @eliasnaur?

Could there be an opt-in flag that we could set to force the Gio software renderer? Or maybe there already is?

**Comment by danderson** [comment]
Can also reproduce on an emulated Pixel 3a, API 30. The emulated device in general works fine, but Tailscale just pulls up a black screen.

**Comment by danderson** [comment]
From `adb logcat com.tailscale.ipn:I '*:S'`, during app startup:

```
12-16 20:05:32.970 19196 19241 I com.tailscale.ipn: fatal error: no support for OpenGL ES 3 nor EXT_sRGB
```

Looks like the emulator doesn't support the graphics API that gio tries to use?

That's the only log entry for the Tailscale app that's not generic all-plaforms Tailscale stuff.

**Comment by danderson** [comment]
Aha! Fixed it. With the emulator running:

 - Click three-dots menu to open emulator settings
 - Go to Settings, Advanced
 - Switch "OpenGL ES API level (requires restart)" from "Autoselect" to "Renderer maximum (up to OpenGL ES 3.1)"
 - Quit the emulator
 - In Android Studio, select the emulated device, hit the dropdown arrow, and "Cold boot now".

Once Android boots back up, the Tailscale GUI renders correctly.

**Comment by danderson** [comment]
With more words: the other OpenGL ES option you can select is "Compatibility (OpenGL ES 1.1/2.0)". I'm assuming the emulator autoselects that, and so then gio is sad because the emulator doesn't give it the OpenGL version it wants.

**Comment by DentonGentry** [comment]
Setting OpenGL 3.1 for the emulator works for me as well, the Tailscale app now renders correctly where before it was a black screen.

**Comment by eliasnaur** [comment]
Closing as duplicate of #1008; when that is fixed, the OpenGL 3.1 workaround will no longer be necessary for the emulator.
